game-core/src/main/java/games/strategy/triplea/odds/calculator/ConcurrentBattleCalculator.java

```diff
@@ -223,44 +223,41 @@ public AggregateResults calculate(
       final Collection<Unit> bombarding,
       final Collection<TerritoryEffect> territoryEffects,
       final boolean retreatWhenOnlyAirLeft,
-      final int initialRunCount)
+      final int runCount)
       throws IllegalStateException {
     synchronized (mutexCalcIsRunning) {
       awaitLatch();
       final long start = System.currentTimeMillis();
+      final AtomicInteger overflowRunsAvailable = new AtomicInteger(runCount % workers.size());
       // Create worker thread pool and start all workers
-      int totalRunCount = 0;
-      int runCount = initialRunCount;
-      final int workerNum = workers.size();
-      final int workerRunCount = Math.max(1, (runCount / Math.max(1, workerNum)));
-      final List<Future<AggregateResults>> list = new ArrayList<>();
-      for (final BattleCalculator worker : workers) {
-        if (!getIsReady()) {
-          // we could have attempted to set a new game data, while the old one was still being set,
-          // causing it to abort with null data
-          return new AggregateResults(0);
-        }
-        final int currentWorkedRunCount = (runCount <= 0 ? 0 : workerRunCount);
-        if (currentWorkedRunCount > 0) {
-          totalRunCount += currentWorkedRunCount;
-          list.add(
-              executor.submit(
-                  () ->
-                      worker.calculate(
-                          attacker,
-                          defender,
-                          location,
-                          attacking,
-                          defending,
-                          bombarding,
-                          territoryEffects,
-                          retreatWhenOnlyAirLeft,
-                          currentWorkedRunCount)));
-        }
-        runCount -= workerRunCount;
+      final int runsPerWorker = runCount / workers.size();
+      final List<Future<AggregateResults>> list =
+          workers.stream()
+              .map(
+                  worker ->
+                      executor.submit(
+                          () ->
+                              worker.calculate(
+                                  attacker,
+                                  defender,
+                                  location,
+                                  attacking,
+                                  defending,
+                                  bombarding,
+                                  territoryEffects,
+                                  retreatWhenOnlyAirLeft,
+                                  // Ensure that we always achieve the target run count even if
+                                  // the number is not dividable by workers.size()
+                                  (overflowRunsAvailable.getAndDecrement() > 0 ? 1 : 0)
```

Comment:
Question: I wonder if there is a way to make this a bit faster/easier to understand?

Suggestion: Have you considered extracting this bit of logic to a method perhaps? In that case the bit of logic could be labelled and perhaps be commented.

Reply:
I have thought about extracting the logic into a method, but you'd still have to pass an `AtomicInteger` instance to the method because it has to be shared for the same call.

However I think extracting this "distribution" to an object and adding unit tests for that could be an elegant abstraction.
I won't have the time to do this today, this will have to wait a couple of days.